### PR TITLE
feat: Include version number in all --json based outputs

### PR DESCRIPTION
**Pull request by kamilogorek**
Affected commands:
- `deno doc --json`
- `deno info --json`
- `deno lint --json`
- `deno lint --json --rules`
- `deno bench --json`

Things to note:
- `[WILDCARD]` could be replaced with `[WILDCARD].[WILDCARD].[WILDCARD]` for more accurate assertion
- *breaking:* `doc --json` returns the output under `nodes` now
- *breaking:* `lint --json` returns the output under `rules` now
- missing `bench --json` test, as I couldn't get it running 🤔 

Resolves https://github.com/denoland/deno/issues/17774

### CODE REVIEW

File: cli/tools/doc.rs [hunk 1]
```diff
@@ -87,7 +88,11 @@ pub async fn print_docs(
   };
 
   if doc_flags.json {
-    write_json_to_stdout(&doc_nodes)
+    let json_output = serde_json::json!({
+      "version": version::deno(),
```
Comment (dsherret) [inline cli/tools/doc.rs:92]:
```suggestion
      "version": 1,
```

So not the deno version, but a version that signifies the schema version of this json. Then we can bump this number whenever it changes. So we can start with `1` for all of these.

File: cli/tests/testdata/doc/json.out [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "version": "[WILDCARD]",
```
Comment (dsherret) [inline cli/tests/testdata/doc/json.out:2]:
```suggestion
  "version": 1,
```

See other comment.

**Comment by kamilogorek** [comment]
Ah, gotcha. Thought that maybe detecting deno version that generated the output is sufficient. Updated.

**Comment by aapoalas** [comment]
Breaking changes in the `--json` commands are, well, breaking. Neither `lint --json` nor `doc --json` are marked as unstable anymore so at least formally this should be a 2.0 thing.

**Comment by dsherret** [comment]
Yes, the original issue was for 2.0.

**Comment by aapoalas** [comment]
Needs rebase, and then of course have to wait for a major version change :)

**Comment by kamilogorek** [comment]
Ping me once it's ready for 2.0, so I don't have to rebase multiple times :)

**Comment by bartlomieju** [comment]
@kamilogorek is there a particular reason you deleted the fork repo? We're on the verge of Deno 2 and wanted to merge this PR. Should we open a new one then?

**Comment by kamilogorek** [comment]
I might've gone too far pruning stale PRs on my GitHub account 😅 resubmitted as https://github.com/denoland/deno/pull/25335

**Pull request closed without merging**